### PR TITLE
Medication post route

### PR DESCRIPTION
**Pull request by Urszuja**
no test provided

### CODE REVIEW

File: backend/src/middlewares/medicationMiddlewares.js [hunk 1]
```diff
@@ -31,3 +31,34 @@ async function deleteMedication(req, res, next) {
...
+    nameOfMedication: medicationData.nameOfMedication,
+    quantity: medicationData.quantity,
+    addDate: medicationData.addDate,
+    dosage: medicationData.dosage,
+    category: medicationData.category,
+    expirationDate: medicationData.expirationDate,
+    profile: medicationData.profile,
+  });
+  await medication.save();
+  res.status(StatusCodes.OK).json(medication);
```
Comment (Nefariusek) [inline backend/src/middlewares/medicationMiddlewares.js:63]: remove

[the remark this answers is not in the document]
Comment (Nefariusek) [reply backend/src/middlewares/medicationMiddlewares.js:63]: const newMedication = await medication.save();

[the remark this answers is not in the document]
Comment (Nefariusek) [reply backend/src/middlewares/medicationMiddlewares.js:63]: const newMedication = await medication.save();

File: backend/src/middlewares/medicationMiddlewares.js [hunk 2]
```diff
@@ -31,3 +31,34 @@ async function deleteMedication(req, res, next) {
...
+    res.status(StatusCodes.CREATED).json(newMedication);
+  } catch (err) {
+    next(err);
+  }
+}
+
+export { postMedication };
+
+async function addNewMedication(medicationData) {
+  const existingMedication = await Medication.findOne({ nameOfMedication: medicationData.nameOfMedication });
```
Comment (Nefariusek) [inline backend/src/middlewares/medicationMiddlewares.js:47]: check for duplicates for single profile (we have many users with the same medication name)

Use medication.profile as an reference to Profile

Comment (Nefariusek) [inline backend/src/middlewares/medicationMiddlewares.js:47]: check for duplicates for single profile (we have many users with the same medication name)

Use medication.profile as an reference to Profile

Comment (Nefariusek) [reply backend/src/middlewares/medicationMiddlewares.js:47]: We expect to send a request with id of profile in it

File: backend/src/middlewares/userMiddlewares.js [hunk 3]
```diff
@@ -106,3 +106,24 @@ async function registerNewUser(userData) {
...
+    quantity,
+    addDate,
+    dosage,
+    category,
+    expirationDate,
+    profile,
+  });
+  await medication.save();
+  res.status(StatusCodes.OK).json(medication);
+}
```
Comment (Nefariusek) [inline backend/src/middlewares/userMiddlewares.js:129]: remove it from this file

Comment (Nefariusek) [inline backend/src/middlewares/userMiddlewares.js:129]: remove it from this file

File: backend/src/middlewares/medicationMiddlewares.js [hunk 4]
```diff
@@ -31,3 +31,34 @@ async function deleteMedication(req, res, next) {
...
+
+  const medication = new Medication({
+    nameOfMedication: medicationData.nameOfMedication,
+    quantity: medicationData.quantity,
+    addDate: medicationData.addDate,
+    dosage: medicationData.dosage,
+    category: medicationData.category,
+    expirationDate: medicationData.expirationDate,
+    profile: medicationData.profile,
+  });
```
Comment (Nefariusek) [inline backend/src/middlewares/medicationMiddlewares.js:61]: you can spread body of request

Comment (Nefariusek) [inline backend/src/middlewares/medicationMiddlewares.js:61]: you can spread body of request

**Review by Nefariusek — CHANGES_REQUESTED** [review]

File: backend/src/middlewares/userMiddlewares.js (hunk 3 above)
Comment (Urszuja) [reply backend/src/middlewares/userMiddlewares.js:129]: ok

File: backend/src/middlewares/medicationMiddlewares.js (hunk 4 above)
Comment (Urszuja) [reply backend/src/middlewares/medicationMiddlewares.js:61]: ok

Comment (Urszuja) [reply backend/src/middlewares/medicationMiddlewares.js:61]: ok

File: backend/src/middlewares/userMiddlewares.js (hunk 3 above)
Comment (Urszuja) [reply backend/src/middlewares/userMiddlewares.js:129]: ok

File: backend/src/middlewares/medicationMiddlewares.js [hunk 5]
```diff
@@ -31,3 +31,25 @@ async function deleteMedication(req, res, next) {
...
+      nameOfMedication: req.body.nameOfMedication,
+    });
+    if (existingMedication) {
+      return new ExpressError('Medication already in your aidkit', StatusCodes.CONFLICT);
+    } else {
+      const medication = new Medication({
+        ...req.body,
+      });
+      const addedMedication = await medication.save();
+      return addedMedication;
```
Comment (Nefariusek) [inline backend/src/middlewares/medicationMiddlewares.js:48]: We should send a response with added medication instead of returning this document

**Review by Nefariusek — CHANGES_REQUESTED** [review]

**Review by Nefariusek — APPROVED** [review]

**Pull request merged**